### DISCUSSION TITLE
Support for x-field-extra-annotation

### DISCUSSION THREAD

**Discussion by albertopneto** [quarkiverse/quarkus-openapi-generator#838] [General]
Hi folks, I am trying to add a custom annotation for a property that exists in a DTO class, however I'm no able to make the x-field-extra-annotation to work.

that x-field-extra-annotation works fine when using open api generator with maven plugin.

Seems that the quarkus-generator-api ignores it...

is that a known issue ?

**Comment by ricardozanini** [comment]
It's not an issue, but a feature we haven't implemented yet. Feel free to send a PR!

**Comment by lordvlad** [comment]
Is there any way to easily hook into the generation process? Or will the PR require a generic Hook mechanism to process extensions?

**Reply by lordvlad** [reply]
Looking at these, it feels like in `pojo.qute` at least, it should have been `x-field-extra-annotations`, or am I wrong? 

https://github.com/search?q=repo%3Aquarkiverse%2Fquarkus-openapi-generator%20x-extra-annotation&type=code

**Reply by lordvlad** [reply]
I can confirm it works with `x-extra-annotation` instead of `x-extra-field-annotation`

**Reply by ricardozanini** [reply]
@lordvlad, thanks for confirming it. I'm closing this discussion as answered.

**Comment by lordvlad** [comment]
But is it though? Shouldn't it be changed to accept
`x-field-extra-annotations` instead?

Ricardo Zanini ***@***.***> schrieb am Mo., 14. Apr. 2025,
18:01:

**Comment by ricardozanini** [comment]
@lordvlad Not sure. What tells the openapi-generator docs? If it's `x-field-extra-annotations`, so yes you may send a PR changing it. 👍

**Reply by lordvlad** [reply]
If I read [this](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/JavaSpring/pojo.mustache) right, `x-extra-annotation` goes on the getter, while `x-field-extra-annotation` goes on the field itself, so technically, both extensions could be applied separately. For my use-case, I actually need it on the getter, so current implementation is sufficient. Sorry for the stir-up

**Reply by ricardozanini** [reply]
No worries, thanks for bringing this in.
